src/runner.rs

```diff
@@ -221,10 +232,11 @@ impl Interpreter {
 		Ok(opt_return_value)
 	}
 
-	pub fn resume_execution<'a, E: Externals + 'a>(
+	pub(crate) fn resume_execution<'a, E: Externals + 'a>(
 		&mut self,
 		return_val: Option<RuntimeValue>,
 		externals: &'a mut E,
+		return_type: Option<ValueType>,
```

Comment:
Shouldn't we store this in the `Interpreter` instead of forcing the user to pass it in? It just opens the door to logic errors.

Reply:
Yes.

This will be easiest to implement when I change Interpreter to be type safe.